src/ja/common/job.py

```diff
@@ -0,0 +1,103 @@
+'''
+This module contains the definition of the Job class and related structures.
+'''
+from enum import Enum
+from docker_context import DockerConstraints, IDockerContext
+
+
+class JobStatus(Enum):
+    '''
+    Represents the status of a job.
+    '''
+    NEW = 0
+    QUEUED = 1
+    RUNNING = 2
+    PAUSED = 3
+    DONE = 4
```

Comment:
Are we always going to instantly remove the job from the worker client when it's done?

Reply:
What do you mean by remove?

Reply:
I was thinking about SSH into a container that is done and if we would need an extra state to differentiate if the container still being loaded on the machine.
But I think we decided in the meeting that users should only be allowed to SSH into running containers.
Sorry, my bad.